securedrop_client/queue.py

```diff
@@ -56,18 +56,23 @@ def _process(self, exit_loop: bool) -> None:
 
             try:
                 job._do_call_api(self.api_client, session)
-            except RequestTimeoutError:
-                # Reset number of remaining attempts for this job to the default
-                job.remaining_attempts = DEFAULT_NUM_ATTEMPTS
+            except (RequestTimeoutError, SendReplyJobTimeoutError):
+                logger.debug('Job {} timed out'.format(job))
 
-                # Resubmit job without modifying counter to ensure jobs with equal
+                # Reset number of remaining attempts for this job to the default and
+                # resubmit job without modifying counter to ensure jobs with equal
                 # priorities are processed in the order that they were submitted
                 # _by the user_ to the queue.
+                job.remaining_attempts = DEFAULT_NUM_ATTEMPTS
                 self.queue.put_nowait((priority, job))
             except ApiInaccessibleError:
-                # This is a guard against #397, we should pause the queue execution when this
-                # happens in the future and flag the situation to the user (see ticket #379).
+                # This is a guard against #397, we should re-enqueue the job and pause queue
+                # processing when this happens in the future and flag the situation to the user
+                # (see ticket #379).
                 logger.error('Client is not authenticated, skipping job...')
+            except Exception as e:
+                # we should re-enqueue the job and pause queue processing
```

Comment:
let's snip out this comment since generic exceptions we want to just log and continue (reserving pausing for auth error or network issues)

Reply:
whoops yeah this comment liessss